bin/propolis-server/src/lib/stats/process.rs

```diff
@@ -0,0 +1,129 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::io::Read;
+use std::time::{Duration, Instant};
+
+use anyhow::Context;
+use tokio::sync::watch;
+use zerocopy::{AsBytes, FromBytes, FromZeroes};
+
+#[derive(Debug, Default)]
+pub(crate) struct ProcessStats {
+    pub(crate) measurement_time: Duration,
+    pub(crate) rss: usize,
+    pub(crate) vsz: usize,
+}
+
+impl ProcessStats {
+    pub fn new() -> watch::Receiver<ProcessStats> {
+        let (tx, rx) = watch::channel::<ProcessStats>(ProcessStats::default());
+
+        tokio::task::spawn(process_stats_task(tx));
+
+        rx
+    }
+}
+
+const PRFNSZ: usize = 16;
+const PRARGSZ: usize = 80;
+
+/// From `sys/types.h`
+#[allow(non_camel_case_types)]
+type taskid_t = i32;
+/// From `sys/time_impl.h`
+#[allow(non_camel_case_types)]
+type timespec_t = [i64; 2];
+/// From `sys/time_impl.h`
+#[allow(non_camel_case_types)]
+type timestruc_t = timespec_t;
+/// From `sys/types.h`
+#[allow(non_camel_case_types)]
+type dev_t = u64;
+
+/// `psinfo`'s definition depends on the data model reported by illumos. This is
+/// in line with the 64-bit version of the struct, and ignores a few fields at
+/// the end of the struct.
+#[derive(Copy, Clone, Debug, FromZeroes, AsBytes, FromBytes)]
+#[cfg(target_arch = "x86_64")]
```

Comment:
shouldn't the whole module have this `cfg`? none of the code that references `psinfo`, which is ... all of it ... will compile if this isn't there...

Reply:
on the other hand, nothing else here is dependent on the target architecture - i was kind of thinking that given an aarch64 target or something we'd be able to fill in the right struct for `psinfo` on that arch. but, yeah, as-is this just breaks even compiling `propolis-server` on other architectures.

i'll `cfg(target_arch = "x86_64")` psinfo read and have it return 0 with a `warn!()` on other architectures. seeing that would imply running `propolis-server` on non-x86 targets which would be super neat..

Reply:
If memory serves, there's other stuff that will just panic on any other architecture (certainly I believe all the CPUID stuff will), but that seems fine to me!